src/main/java/duke/action/Event.java

```diff
@@ -0,0 +1,74 @@
+package duke.action;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * An extension from Action. The Event class
+ * has a variable date of type LocalDateTime
+ * that is specific to itself.
+ */
+public class Event extends Action {
+
+    private LocalDateTime date;
+
+    /**
+     * Constructs a new Event object with the variables task and at.
+     * @param task event
+     * @param at date and time values of the event
+     */
+    public Event(String task, String at) {
+        super(task);
+        try {
+            DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy-MM-dd H:m");
+            this.date = LocalDateTime.parse(at, format);
+        } catch (DateTimeParseException e) {
+            System.out.println("Wrong date format: Please delete input "
+                    + "and re-enter using yyyy-mm-dd H:m format");
+            this.date = LocalDateTime.now();
+        }
+    }
+
+    /**
+     * Constructs a new Event object with the variables task, date
+     * and bool.
+     * @param task event
+     * @param date date and time values of the event
+     * @param bool done status of the event
+     */
+    public Event(String task, LocalDateTime date, boolean bool){
+        super(task, bool);
+        this.date = date;
+    }
+
+    /**
+     * Returns a new Event object with same variable values except
+     * isDone which is now true
+     * @return marked Event object
+     */
+    @Override
+    public Action setDone()  {
```

Comment:
```suggestion
    public Action setDone() {
```
Coding standard spacing